assets/report_template.Rmd

```diff
@@ -0,0 +1,1445 @@
+flag_filter_len_stats <- nrow(filter_len_stats) > 0
+```
+
+```{r, eval = !isFALSE(params$filter_len_asv) && flag_filter_len_stats, results='asis'}
+# Reads removed
+
+# re-name & re-order columns
+colnames(filter_len_stats) <- gsub("lenfilter_","",colnames(filter_len_stats))
+filter_len_stats <- filter_len_stats[, c("sample", "input", "output")]
+filter_len_stats$'retained%' <- round( filter_len_stats$output / filter_len_stats$input * 100 , 2)
+filter_len_stats_avg_removed <- 100-sum(filter_len_stats$'retained%')/length(filter_len_stats$'retained%')
+filter_len_stats_max_removed <- 100-min(filter_len_stats$'retained%')
+
+cat("The following table shows (read) counts for each sample before and after filtering:")
+
+# Display table
+datatable(filter_len_stats, options = list(
+        scrollX = TRUE,
+        scrollY = "300px",
+        paging = FALSE))
+
+cat("In average", filter_len_stats_avg_removed, "% reads were removed, but at most",filter_len_stats_max_removed,"% reads per sample.")
+```
+
+```{r, eval = !isFALSE(params$filter_len_asv_len_orig), results='asis'}
+cat("The number of ASVs was reduced by",sum(filter_len_profile$Counts)-sum(filter_len_asv_filtered$Counts),"(",100-round( sum(filter_len_asv_filtered$Counts)/sum(filter_len_profile$Counts)*100 ,2),"%), from",sum(filter_len_profile$Counts),"to",sum(filter_len_asv_filtered$Counts)," ASVs.")
+cat("\n\nLength filter results can be found in folder [asv_length_filter](../asv_length_filter).")
+```
+
+<!-- Subsection on codon usage filter -->
+
+```{r, eval = !isFALSE(params$filter_codons), results='asis'}
+cat(paste0("
+## Codon usage
+
+Amplicons of coding regions are expected to be free of stop codons and consist of condon tripletts.
+ASVs were filtered against the presence of stop codons (",params$stop_codons,") in the specified open reading frame of the ASV.
+Additionally, ASVs that are not multiple of 3 in length were omitted.
+
+"))
+
+# import stats tsv
+filter_codons_stats <- read.table(file = params$filter_codons, header = TRUE, sep = "\t")
+
+cat("The following table shows read counts for each sample after filtering:")
+
+# Display table
+datatable(filter_codons_stats, options = list(
+        scrollX = TRUE,
+        scrollY = "300px",
+        paging = FALSE))
+
+#TODO: add ASV count after filtering
+
+cat("\n\nCodon usage filter results can be found in folder [codon_filter](../codon_filter).")
+```
+
+<!-- Section on taxonomic classification -->
+
+```{r, results='asis'}
+# Check if any taxonomic classification is available
+any_taxonomy <- !isFALSE(params$dada2_taxonomy) || !isFALSE(params$qiime2_taxonomy) || !isFALSE(params$sintax_taxonomy) || !isFALSE(params$pplace_taxonomy)
+```
+
+```{r, eval = any_taxonomy, results='asis'}
+# Header if any taxonomic classification is available
+cat("# Taxonomic Classification\n")
+```
+
+<!-- Subsection on ITS region filter -->
+
+```{r, eval = !isFALSE(params$cut_its), results='asis'}
+cat(paste0("
+## ITS regions
```

Comment:
Based on the discussion in https://nfcore.slack.com/archives/CEA7TBJGJ/p1690893776838869 we could add at least a few more sentences I think.